test/TEST-12-ISSUE-3171/test.sh

```diff
@@ -6,8 +6,12 @@ TEST_DESCRIPTION="https://github.com/systemd/systemd/issues/3171"
 . $TEST_BASE_DIR/test-functions
 
 test_run() {
-    run_nspawn || return 1
-    check_result_nspawn || return 1
+    if check_nspawn; then
+        run_nspawn
+        check_result_nspawn || return 1
+    else
+        dwarn "can't run systemd-nspawn, skipping"
+    fi
```

Comment:
So, we have:
```
    if run_qemu; then
        check_result_qemu || return 1
```
and
```
    if check_nspawn; then
        run_nspawn
        check_result_nspawn || return 1
```
I think we should merge `check_nspawn` to `run_nspawn`. But this is another PR